File: library-service/src/main/java/com/smngs/library/entity/Book.java
package com.smngs.library.entity;

import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "book")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "front_picture")
    private String frontPicture;

    @ManyToMany
    @JoinTable(
            name = "book_genre",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_id"))
    @ToString.Exclude
    private List<Genre> genres;

    @ManyToOne
    @JoinColumn(name = "publisher_id")
    private Publisher publisher;

    @ManyToMany
    @JoinTable(
            name = "book_author",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "author_id"))
    @ToString.Exclude
    private List<Author> authors;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @CreationTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: client/src/types/book.ts
import {BookGenreWithoutBooksResponse} from "./bookCategory";
import {AuthorWithoutBooksResponse} from "./bookAuthor";
import {BookPublisherWithoutBooksResponse} from "./bookPublisher";

export interface BookRequest {
    title: string,
    genres: number[],
    publisher: BookPublisherWithoutBooksResponse,
    authors: number[],
    frontPicture: File
}

export interface BookResponse {
    id: number,
    title: string,
    genres: BookGenreWithoutBooksResponse[],
    publisher: BookPublisherWithoutBooksResponse,
    authors: AuthorWithoutBooksResponse[],
    frontPicture: string
}